components/eamxx/src/share/util/scream_combine_ops.hpp

```diff
@@ -98,6 +100,74 @@ void combine (const ScalarIn& newVal, ScalarOut& result,
       break;
   }
 }
+/* Special version of combine that takes a mask into account */
+template<CombineMode CM, typename ScalarIn, typename ScalarOut,
+         typename CoeffType = typename ekat::ScalarTraits<ScalarIn>::scalar_type>
+KOKKOS_FORCEINLINE_FUNCTION
+void combine_and_fill (const ScalarIn& newVal, ScalarOut& result, const float fill_val,
+              const CoeffType alpha = CoeffType(1),
+              const CoeffType beta  = CoeffType(0))
+{
+  switch (CM) {
+    case CombineMode::Replace:
+      combine<CM>(newVal,result,alpha,beta);
+      break;
+    case CombineMode::Rescale:
+      if (result != fill_val) {
+        combine<CM>(newVal,result,alpha,beta);
+      }
+      break;
+    case CombineMode::ScaleReplace:
+      if (newVal == fill_val) {
+        result = fill_val;
+      } else {
+        combine<CM>(newVal,result,alpha,beta);
+      }
+      break;
+    case CombineMode::Update:
```

Comment:
The other cases are all the same, so you could do
```
	case CombineMode::Update:
	case CombineMode::ScaleUpdate:
	...
	case CombineMode::Divide:
	if (result == fill_val || newVal == fill_val) {
		result = fill_val;
	} else {
		combine<CM>(newVal,result,alpha,beta);
	}
```
which makes it simpler to read.